File: app/src/main/java/com/ytempest/wanandroid/base/activity/LoaderActivity.java
package com.ytempest.wanandroid.base.activity;

import android.view.ViewGroup;

import com.ytempest.wanandroid.base.load.Loader;
import com.ytempest.wanandroid.base.presenter.IPresenter;

/**
 * @author ytempest
 * @since 2021/1/6
 */
public abstract class LoaderActivity<Presenter extends IPresenter> extends MvpActivity<Presenter> {
    private static final String TAG = "LoaderActivity";

    private Loader mLoader;

    @Override
    protected void onViewCreated() {
        super.onViewCreated();
        ViewGroup root = findViewById(android.R.id.content);
        mLoader = new Loader(root);
        mLoader.setReloadCall(aVoid -> onReloadClick());
    }

    protected void onReloadClick() {

    }

    public Loader getLoader() {
        return mLoader;
    }
}

File: app/src/main/java/com/ytempest/wanandroid/base/activity/LoaderActivity.kt
package com.ytempest.wanandroid.base.activity

import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import com.ytempest.wanandroid.base.load.Loader
import com.ytempest.wanandroid.base.presenter.IPresenter

/**
 * @author heqidu
 * @since 21-2-8
 */
abstract class LoaderActivity<Presenter : IPresenter, VB : ViewBinding> : MvpActivity<Presenter, VB>() {
    private lateinit var mLoader: Loader

    override fun onViewCreated() {
        super.onViewCreated()
        val root = findViewById<ViewGroup>(android.R.id.content)
        mLoader = Loader(root)
        mLoader.setReloadCall { onReloadClick() }
    }

    protected open fun onReloadClick() {

    }

    fun getLoader(): Loader {
        return mLoader
    }
}